GenderPayGap.WebUI/Controllers/RegisterController.Request.cs

```diff
@@ -686,6 +686,12 @@ public async Task<IActionResult> ConfirmCancellation(OrganisationViewModel model
                     $"Unused organisation {userOrg.OrganisationId}:'{userOrg.Organisation.OrganisationName}' deleted by {(OriginalUser == null ? currentUser.EmailAddress : OriginalUser.EmailAddress)} when declining manual registration for {userOrg.User.EmailAddress}");
                 DataRepository.Delete(userOrg.Organisation);
             }
+            if (userOrg.Organisation != null
+                && userOrg.Organisation.UserOrganisations.Count == 1
+                && userOrg.Organisation.Status == OrganisationStatuses.Pending)
+            {
+                userOrg.Organisation.SetStatus(OrganisationStatuses.Deleted, details: "Manually Rejected");

```

Comment:
Random question, but why do we have the organisation in the database if they haven't been registered already? Actually what exactly are they registering for?

Reply:
when adding an organisation to their account a user (the user does that so that they can report for that org) can either:
* select an existing organisation (that is already in our db OR is fetched from Companies House) OR
* add a new organisation - the organisation details being provided by the user
In the second case, the organisation is added to the database but its status is Pending. All the other organisations are Active

All the registrations need to be approved by an admin - if the registration is rejected the Pending organisations should be deleted so that they can't be used by other users

Reply:
Is the soft delete for posterity's sake or because the org can be re-activated at a later date?

Reply:
for posterity's sake - I mean it can be reactivated as well but I don't think this will happen

Reply:
Ah cool, sorry I thought I had approved this!